zanata-war/src/main/java/org/zanata/dao/DatabaseSpecific.java

```diff
@@ -0,0 +1,18 @@
+package org.zanata.dao;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marker annotation to mark methods that will only work in certain database.
+ */
+@Target({ElementType.METHOD, ElementType.LOCAL_VARIABLE, ElementType.TYPE, ElementType.PARAMETER})
+@Retention(RetentionPolicy.SOURCE)
+public @interface DatabaseSpecific {
```

Comment:
This is mainly to indicate the query uses some database specific functions. NativeQuery alone can still be portable between different databases.